ui/constants/capabilities.js

```diff
@@ -0,0 +1,39 @@
+export const capabilties = {
+  SyncPrefs : "sync-prefs", // /user/preferences
```

Comment:
Please don't hard code these in UI. it would lead to all kind of inconsistencies.

You can directly check for the presence of `"meshery-patterns-catalog"` this string in the `capabilityRegistry?.capabilities`.  Then based on that boolean value you set the `canPublish` var and then perform the remaining checks as you did.

Reply:
Sure @theBeginner86 , I think I'm late on this PR update, but will do the changes asap.